bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,234 @@
+
+    public YIOremoteDockHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+        config = getConfigAs(YIOremoteConfiguration.class);
+
+        // set the thing status to UNKNOWN temporarily and let the background task decide for the real status.
+        // the framework is then able to reuse the resources from the thing handler initialization.
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.UNKNOWN);
+        try {
+            logger.debug("Starting generating URI_yiodockwebsocketaddress");
+            URI_yiodockwebsocketaddress = new URI("ws://" + config.yiodockhost + ":946");
+            logger.debug("Finished generating URI_yiodockwebsocketaddress");
+            YIOREMOTEHANDLESTATUS_actualstatus = YIOREMOTEHANDLESTATUS.AUTHENTICATION_PROCESS;
+            try {
+
+                logger.debug("Starting websocket Client");
+                YIOremote_DockwebSocketClient.start();
+                logger.debug("Started websocket Client");
+            } catch (Exception e) {
+                logger.warn("Web socket start failed {}", e.toString());
+                // throw new IOException("Web socket start failed");
+            }
+            try {
+                logger.debug("Connect websocket client");
+                YIOremote_DockwebSocketClient.connect(YIOremoteDockwebSocket_ClientSocket, URI_yiodockwebsocketaddress,
+                        YIOremote_DockwebSocketClientrequest);
+                logger.debug("Connected websocket client");
+
+                logger.debug("Check for authentication requested by YIO Dock");
+                YIOremoteDockwebSocket_ClientSocket.getLatch().await();
+                Thread.sleep(1000);
+
+                try {
+                    if (YIOremoteDockwebSocket_ClientSocket.get_boolean_authentication_required()) {
+                        logger.debug("send authentication to YIO dock");
+                        YIOremoteDockwebSocket_ClientSocket.sendMessage(YIOREMOTEMESSAGETYPE.AUTHENTICATE,
+                                config.yiodockaccesstoken);
+                        Thread.sleep(1000);
+
+                        if (YIOremoteDockwebSocket_ClientSocket.get_boolean_authentication_ok()) {
+                            YIOREMOTEHANDLESTATUS_actualstatus = YIOREMOTEHANDLESTATUS.AUTHENTICATED;
+
+                        } else {
+                            logger.debug("authentication to YIO dock not ok");
+                            YIOREMOTEHANDLESTATUS_actualstatus = YIOREMOTEHANDLESTATUS.AUTHENTICATED_FAILED;
+                        }
+                    } else {
+                        logger.debug("authentication error YIO dock");
+                    }
+
+                } catch (IllegalArgumentException e) {
+                    logger.warn("JSON convertion failure {}", e.toString());
```

Comment:
Better use the exception's message. Same for the others.
```suggestion
                    logger.warn("JSON convertion failure {}", e.getMessage());
```